pkg/iac/terraform/state/enumerator/s3.go

```diff
@@ -15,19 +16,48 @@ import (
 )
 
 type S3EnumeratorConfig struct {
-	Bucket *string
-	Prefix *string
+	prefix     string
+	pattern    string
+	defaultEnv map[string]string
 }
 
 type S3Enumerator struct {
 	config config.SupplierConfig
 	client s3iface.S3API
 }
 
+func NewS3EnumeratorConfig() *S3EnumeratorConfig {
+	envMap := map[string]string{}
+	for _, variable := range os.Environ() {
+		tmp := strings.SplitN(variable, "=", 2)
+		envMap[tmp[0]] = tmp[1]
+	}
+	return &S3EnumeratorConfig{
+		prefix:     "DCTL_S3_",
+		pattern:    "AWS_",
+		defaultEnv: envMap,
+	}
+}
+
+func (s *S3EnumeratorConfig) setEnv(resetMod bool) {
+	for key, value := range s.defaultEnv {
+		if strings.HasPrefix(key, s.prefix) {
+			key = strings.Replace(key, s.prefix, s.pattern, 1)
+			if resetMod {
+				value = s.defaultEnv[key]
+			}
+			os.Setenv(key, value)
+		}
+	}
+}
+
 func NewS3Enumerator(config config.SupplierConfig) *S3Enumerator {
+	s3EnumConfig := NewS3EnumeratorConfig()
+	s3EnumConfig.setEnv(false)
```

Comment:
I think you should set the prefix and destination prefix here, since it's directly related to this enumerator.

example (pseudo code)

```go
func NewS3Enumerator(p EnvProxy, config config.SupplierConfig) *S3Enumerator {
  p.SetProxy("DCTL_S3_", "AWS_")

  // ...
```

Reply:
yes, good point, that was my first implementation, i didn't think about reuse that logic